testsuite/do_tests.sh

```diff
@@ -0,0 +1,684 @@
+    fi
+
+    if test "$1" = 2 ; then
+        echo "Error: Missing required option \'$2\'"
+    fi
+
+    if test "$1" = 3 ; then
+        echo "$2"
+    fi
+
+    cat <<EOF
+Usage: do_tests.sh --prefix=<path> --report-dir=<path> [options]"
+
+    --prefix=<path>        The base installation path of NEST
+    --report-dir=<path>    The directory to store the output in
+
+Options:
+
+    --with-python=<exe>    The Python executable to use
+    --python-path=<path>   The PYTHONPATH for the NEST installation
+    --with-music=<exe>     The MUSIC executable to use
+    --help                 Print program options and exit
+EOF
+
+    exit "$1"
+}
+
+
+#
+# bail_out message
+#
+bail_out ()
+{
+    echo "$1"
+    exit 1
+}
+
+
+#
+# sed has different syntax for extended regular expressions
+# on different operating systems:
+# BSD: -E
+# other: -r
+#
+EXTENDED_REGEX_PARAM="r"
+/bin/sh -c "echo 'hello' | sed -${EXTENDED_REGEX_PARAM} 's/[aeou]/_/g' "  >/dev/null 2>&1 || EXTENDED_REGEX_PARAM=E
+
+
+
+RUN_TEST="$(dirname $0)/run_test"
+MUSIC=""
+PYTHON=""
+
+
+while test $# -gt 0 ; do
+    case "$1" in
+        --help)
+            usage 0
+            ;;
+        --prefix=*)
+            PREFIX="$( echo "$1" | sed 's/^--prefix=//' )"
+	    if test ! "${PREFIX}"; then usage 2 "--prefix"; fi
+            ;;
+        --report-dir=*)
+            REPORTDIR="$( echo "$1" | sed 's/^--report-dir=//' )"
+	    if test ! "${REPORTDIR}"; then usage 2 "--report-dir"; fi
+            ;;
+        --with-python=*)
+            PYTHON="$( echo "$1" | sed 's/^--with-python=//' )"
+            ;;
+        --python-path=*)
+            PYTHONPATH_="$( echo "$1" | sed 's/^--python-path=//' )"
+            ;;
+        --with-music=*)
+            MUSIC="$( echo "$1" | sed 's/^--with-music=//' )"
+            ;;
+        *)
+            usage 1 "$1"
+            ;;
+    esac
+    shift
+done
+if test ! "${PREFIX}"; then usage 2 "--prefix"; fi
+
+export NEST="nest_serial"
+export PATH="${PREFIX}/bin:$PATH"
+
+unset NEST_INSTALL_DIR
+unset NEST_DATA_DIR
+unset NEST_DOCL_DIR
+if test "${PYTHON}"; then
+    if test ! "${PYTHONPATH_}"; then
+	usage 3 "Error: \'--with-python\' also requires \'--python-path\'"
+    fi
+
+    NOSETESTS="$(command -v nosetests 2>&1)"
+    PYTHON_HARNESS="${PREFIX}/share/nest/extras/do_tests.py"
+
+    export PYTHONPATH="$PYTHONPATH_:${PYTHONPATH:-}"
+fi
+
+export TEST_BASEDIR="${PREFIX}/share/doc/nest"
+
+# Gather some information about the host
+INFO_ARCH="$(uname -m)"
+INFO_HOME="$(/bin/sh -c 'echo ~')"
+INFO_HOST="$(hostname -f)"
+INFO_OS="$(uname -s)"
+INFO_USER="$(whoami)"
+INFO_VER="$(uname -r)"
+NPROCS="$(cat /proc/cpuinfo | grep processor | wc -l)"
+
+print_paths () {
+    indent="`printf '%23s'`"
+    echo "$1" | sed "s/:/\n$indent/g" | sed '/^\s*$/d'
+}
+
+echo "================================================================================"
+echo
+echo "  NEST testsuite"
+echo "  Date: $(date -u)"
+echo "  Sysinfo: $(uname -s -r -m -o)"
```

Comment:
`uname`  on OSX does not support `-o`, I'd suggest to remove it.

Reply:
Would `-a` work on OSX?
```suggestion
echo "  Sysinfo: $(uname -a)"
```
I'm not sure if there was a reason against that…